### PR TITLE
Fix slice tool doesn't work correctly in a tilemap layer when we are in Manual mode (fix #4290)

### LINKED ISSUE #4290: Slice tool doesn't work correctly in a tilemap layer when we are in Manual mode
Report: https://community.aseprite.org/t/tilemap-and-slice/21342

### PR DESCRIPTION
**Pull request by Gasparoken**
fix #4290

### CODE REVIEW

File: src/app/ui/editor/tool_loop_impl.cpp [hunk 1]
```diff
@@ -530,10 +533,10 @@ class ToolLoopImpl : public ToolLoopBase,
         (m_layer->isTilemap() &&
          site.tilemapMode() == TilemapMode::Pixels &&
          site.tilesetMode() == TilesetMode::Manual &&
-         !m_ink->isSelection() ? ExpandCelCanvas::TilesetPreview:
-                                 ExpandCelCanvas::None) |
-        (m_ink->isSelection() ? ExpandCelCanvas::SelectionPreview:
-                                ExpandCelCanvas::None))));
+         !isSelectionPreview ? ExpandCelCanvas::TilesetPreview:
+                               ExpandCelCanvas::None) |
```
Comment (martincapello) [inline src/app/ui/editor/tool_loop_impl.cpp:537]: Should we use parenthesis here?
```suggestion
         (!isSelectionPreview ? ExpandCelCanvas::TilesetPreview:
                                ExpandCelCanvas::None)) |
```
I know that the previous code didn't have them, but just wondering if it was wrong as well.

**Review by martincapello — COMMENTED** [review]
It works! I left just one minor comment.

File: src/app/ui/editor/tool_loop_impl.cpp (hunk 1 above)
Comment (Gasparoken) [reply src/app/ui/editor/tool_loop_impl.cpp]: Good to avoid confusion. Done.

Comment (dacap) [reply src/app/ui/editor/tool_loop_impl.cpp]: This has changed the condition and the result of the operation (go back to the previous version):
```
        (m_layer->isTilemap() &&
         site.tilemapMode() == TilemapMode::Pixels &&
         site.tilesetMode() == TilesetMode::Manual &&
         !isSelectionPreview ? ExpandCelCanvas::TilesetPreview:
                                 ExpandCelCanvas::None) |
```
is not equal to:
```
        (m_layer->isTilemap() &&
         site.tilemapMode() == TilemapMode::Pixels &&
         site.tilesetMode() == TilesetMode::Manual &&
         (!isSelectionPreview ? ExpandCelCanvas::TilesetPreview:
                                ExpandCelCanvas::None)) |
```
The first one returns `ExpandCelCanvas::TilesetPreview` or `ExpandCelCanvas::None` and the second one returns true or false (`1` or `0` in the `|` operation of ints).

Comment (dacap) [reply src/app/ui/editor/tool_loop_impl.cpp]: @Gasparoken I can take this PR and modify the commit directly if you wish (I'm planning to release a new version today if we can)

**Pull request merged**